### ISSUE TITLE
segfault when using function after context goes out of scope

### ISSUE THREAD

**Issue by frozolotl** [TheDan64/inkwell#120]
**Describe the Bug**
The program segfaults when context goes out of scope and the built function is used.

**To Reproduce**
```rs
use inkwell::context::Context;
use inkwell::values::FunctionValue;

fn compile() -> Option<FunctionValue> {
    let context = Context::create();
    let module = context.create_module("example");
    let builder = context.create_builder();

    let fn_type = context.void_type().fn_type(&[], false);
    let function = module.add_function("do_nothing", fn_type, None);
    let basic_block = context.append_basic_block(&function, "entry");

    builder.position_at_end(&basic_block);

    builder.build_return(None);

    Some(function)
}

fn main() {
    let function = compile().unwrap();

    function.print_to_stderr();
}
```

**Expected Behavior**
No crash and some information about the generated code.

**LLVM Version:**
 - LLVM Version: 8.0.1
 - Inkwell Branch Used: llvm8-0

**Desktop:**
 - OS: Linux (5.2.11-arch1-1-ARCH)

**Comment by TheDan64** [comment]
Yeah this might be a dupe of #65 - or at least very closely related to it

**Comment by TheDan64** [comment]
Fixed by 9561e33cd2d4934b69f395b5503b36556f933bb4
